src/physics/sea_ice.jl

```diff
@@ -72,20 +72,22 @@ function sea_ice_timestep!( progn::PrognosticVariables,
     f = sea_ice_model.freeze_rate
     temp_freeze = sea_ice_model.temp_freeze
 
-    # Euler forward step
-    @inbounds for ij in eachgridpoint(sst, ice, mask)
-        if mask[ij] < 1     # at least partially ocean
-
-            # ice-sst flux as a relaxation term wrt to freezing, with different melt/freeze rates
-            dT = sst[ij] - temp_freeze              # uncorrected difference to freezing temperature
-            F = -m*max(dT, 0) - f*min(dT, 0)        # melt if above freezing, freeze if below
-            sst[ij] = max(sst[ij], temp_freeze)     # cap sst at freezing
-
-            # update sea ice concentration, cap between [0, 1]
-            ice[ij] += Δt*F
-            ice[ij] = max(min(ice[ij], 1), 0)
-        end
-    end
+    launch!(architecture(ice), LinearWorkOrder, size(ice), sea_ice_kernel!, ice, sst, mask, temp_freeze, m, f, Δt)
+    synchronize(architecture(ice))
```

Comment:
I don't think you need synchronization after a launch. If you are running on a GPU, all the kernels are executed on one stream so the GPU is synchronized with itself. If you pass memory between the CPU and GPU, the GPU automatically synchronizes. You need to explicitly synchronize only if you are using MPI ad that runs on a different stream. All other explicit synchronizations will lead the performance loss.

Reply:
My understanding so far was that this is the case if you only use KA, but we do use some CUDA.jl functions/executations as well. CUFFT is unavoidable for us, then we also use e.g. some broadcasting here and there (which we could in theory replace). Or is this fine? Then we should really remove also other explicit synchronisation calls we already have.